### PR TITLE
Fix unit test failures during daylight saving time

### PR DESCRIPTION
**Pull request by mattjohnsonpint**
Several unit tests failed as we approached the Spring Forward DST transition.  These were all just testing failures.

### CODE REVIEW

**Comment by mj1856** [comment]
Since this PR touches every locale, I'd like to ask each locale author (or other native speaker) to verify the change for their locale(s).  Note that all of the changes here only touch the unit tests, so the tests have been updated to match what the locale data currently generates.

Most of the locales only had changes to the "calendar day" tests.  Instead of basing the starting moment at 02:00, it was switched to 12:00.  Since daylight saving time transitions usually occur in the early hours of the morning, testing these things in the afternoon will prevent failures near local DST.

Some of the locales (`da`, `dv`, `fo`, `lb`) were missing some tests, so I added them.  Please validate these ones thoroughly.

Please verify the following:
- Do the changes look accurate?
- Does the updated test still exercise enough of the language to validate the `calendar` function?
- Are there any other tests you think should be added?

We will likely merge this PR as-is.  If you need to make additional changes, please send a subsequent PR for your locale.

If all looks good, please comment here with your locale(s) so I can keep track.  Thanks.

- [ ] af @wernerm
- [ ] ar-ma @ElFadiliY @abdelsaid
- [ ] ar-sa @xsoh
- [ ] ar-tn @naderio
- [ ] ar @abdelsaid @forabi
- [ ] az @topchiyev
- [ ] be @demidov91 @Oire
- [ ] bg @kraz
- [ ] bn @kaushikgandhi
- [ ] bo @vajradog
- [ ] br @jbleduigou
- [ ] bs @frontyard
- [ ] ca @juanghurtado
- [ ] cs @petrbela
- [ ] cv @mirontoli
- [ ] cy @robgallen
- [ ] da @mrbase
- [ ] de-at @lluchs @Oire @MadMG @mik01aj
- [ ] de @lluchs @Oire @mik01aj
- [ ] dv @jawish
- [ ] el @mehiel
- [ ] en-au @jarcoal
- [ ] en-ca @jonbca
- [ ] en-gb @chrisgedrim
- [ ] en-ie @chriscartlidge
- [ ] en-nz @lukemcgregor
- [ ] en
- [ ] eo @colindean
- [ ] es @julionc
- [ ] et @madhenry @ragulka
- [ ] eu @eillarra
- [ ] fa @ebraminio
- [ ] fi @bleadof
- [ ] fo @ragnar123
- [ ] fr-ca @jonbca
- [ ] fr-ch @gaspard
- [ ] fr @jfroffice
- [ ] fy @robin0van0der0v
- [ ] gd @jonashdown
- [ ] gl @juanghurtado
- [ ] he @tomer @DevelopmentIL @TalAter
- [ ] hi @mayanksinghal
- [ ] hr @bmarkovic
- [ ] hu @adambrunner
- [ ] hy-am @armendarabyan
- [ ] id @tyok
- [ ] is @hinrik
- [ ] it @aliem @nostalgiaz
- [ ] ja @baryon
- [ ] jv @lantip
- [ ] ka @irakli-janiashvili
- [ ] kk @nurlan
- [ ] km @kruyvanna
- [ ] ko @kyungw00k
- [ ] lb @mweimerskirch @kwisatz
- [ ] lo @ryanhart2
- [ ] lt @mmozuras
- [ ] lv @skakri @JanisE
- [ ] me @miodragnikac
- [ ] mk @B0k0
- [ ] ml @floydpink
- [ ] mr @kalehv @vnathalye
- [ ] ms-my @weldan
- [ ] ms @weldan
- [ ] my @gholadr @thanyawzinmin
- [ ] nb @rexxars @sigurdga
- [ ] ne @suvash
- [ ] nl @jjupiter
- [ ] nn @mechuwind
- [ ] pa-in @harpreetkhalsagtbit
- [ ] pl @evoL
- [ ] pt-br @caio-ribeiro-pereira
- [ ] pt @jalex79
- [ ] ro @gurdiga @avaly
- [ ] ru @Viktorminator @Oire @socketpair
- [ ] se @karamell
- [ ] si @sampathsris
- [ ] sk @k2s @petrbela
- [ ] sl @sedovsek
- [ ] sq @flakerimi @Oire @oerd
- [ ] sr-cyrl @milan-j
- [ ] sr @milan-j
- [ ] sv @ulmus
- [ ] sw @fadsel
- [ ] ta @tk120404
- [ ] te @kcthota
- [ ] th @sirn
- [ ] tl-ph @hagmandan
- [ ] tlh @amaranthrose
- [ ] tr @erhangundogan @BYK
- [ ] tzl @robin0van0der0v
- [ ] tzm-latn @abdelsaid
- [ ] tzm @abdelsaid
- [ ] uk @zemlanin @Oire
- [ ] uz @muminoff
- [ ] vi @bangnk
- [ ] zh-cn @suupic
- [ ] zh-tw @zenozeng

**Comment by mj1856** [comment]
- [ ] id @tyok
- [ ] is @hinrik
- [ ] it @aliem @nostalgiaz
- [ ] ja @baryon
- [ ] jv @lantip
- [ ] ka @irakli-janiashvili
- [ ] kk @nurlan
- [ ] km @kruyvanna
- [ ] ko @kyungw00k
- [ ] lb @mweimerskirch @kwisatz
- [ ] lo @ryanhart2
- [ ] lt @mmozuras
- [ ] lv @skakri @JanisE
- [ ] me @miodragnikac
- [ ] mk @B0k0
- [ ] ml @floydpink
- [ ] mr @kalehv @vnathalye
- [ ] ms-my @weldan
- [ ] ms @weldan
- [ ] my @gholadr @thanyawzinmin
- [ ] nb @rexxars @sigurdga
- [ ] ne @suvash
- [ ] nl @jjupiter
- [ ] nn @mechuwind
- [ ] pa-in @harpreetkhalsagtbit
- [ ] pl @evoL
- [ ] pt-br @caio-ribeiro-pereira
- [ ] pt @jalex79
- [ ] ro @gurdiga @avaly
- [ ] ru @Viktorminator @Oire @socketpair
- [ ] se @karamell
- [ ] si @sampathsris
- [ ] sk @k2s @petrbela
- [ ] sl @sedovsek
- [ ] sq @flakerimi @Oire @oerd
- [ ] sr-cyrl @milan-j
- [ ] sr @milan-j
- [ ] sv @ulmus
- [ ] sw @fadsel
- [ ] ta @tk120404
- [ ] te @kcthota
- [ ] th @sirn
- [ ] tl-ph @hagmandan
- [ ] tlh @amaranthrose
- [ ] tr @erhangundogan @BYK
- [ ] tzl @robin0van0der0v
- [ ] tzm-latn @abdelsaid
- [ ] tzm @abdelsaid
- [ ] uk @zemlanin @Oire
- [ ] uz @muminoff
- [ ] vi @bangnk
- [ ] zh-cn @suupic
- [ ] zh-tw @zenozeng

**Comment by mj1856** [comment]
- [ ] ta @tk120404
- [ ] te @kcthota
- [ ] th @sirn
- [ ] tl-ph @hagmandan
- [ ] tlh @amaranthrose
- [ ] tr @erhangundogan @BYK
- [ ] tzl @robin0van0der0v
- [ ] tzm-latn @abdelsaid
- [ ] tzm @abdelsaid
- [ ] uk @zemlanin @Oire
- [ ] uz @muminoff
- [ ] vi @bangnk
- [ ] zh-cn @suupic
- [ ] zh-tw @zenozeng

**Comment by ulmus** [comment]
sv looks good

**Comment by B0k0** [comment]
mk is good

**Comment by floydpink** [comment]
`ml` looks good. :+1:

**Comment by mehiel** [comment]
`el` is fine as well

**Comment by xsoh** [comment]
`ar-sa` looks good :+1: 

Also I've checked `ar-ma`, `ar-tn ` & `ar`:

- `ar-ma` looks good :+1: 
- `ar-tn` looks good :+1: 
- `ar` looks good :+1:

**Comment by mrbase** [comment]
`da` looks fine

**Comment by socketpair** [comment]
Russian (ru) is good

**Comment by kaushikgandhi** [comment]
bn Looks good

**Comment by zemlanin** [comment]
`uk` is fine

**Comment by baryon** [comment]
ja looks ok

**Comment by chrisgedrim** [comment]
en-gb looks fine to me

**Comment by Karamell** [comment]
`se` is fine.

**Comment by kraz** [comment]
`bg` is fine. :+1:

**Comment by robgallen** [comment]
`cy` looks fine to me.

**Comment by sedovsek** [comment]
`sl` lgtm :+1:

**Comment by petrbela** [comment]
`cs` and `sk` :+1:

**Comment by evoL** [comment]
`pl` is fine :+1:

**Comment by jonbca** [comment]
en-ca and fr-ca checking in. Looks good.

**Comment by Oire** [comment]
`be`, `ru`, and `uk` are fine. Thanks @mj1856!

**Comment by rexxars** [comment]
`nb` looks good.

**Comment by jorisroling** [comment]
nl looks fine

**Comment by BYK** [comment]
`tr` :+1:

**Comment by caio-ribeiro-pereira** [comment]
The `pt-br` is fine!

**Comment by kwisatz** [comment]
Changed and additional tests for `lb` look good.
The _weeks year starting sunday formatted_ tests could use other dates in order to properly test edge cases, but will submit a separate PR for those, as suggested.

**Comment by lantip** [comment]
`jv` looks good. thanks

**Comment by ebraminio** [comment]
`fa` LGTM

**Comment by colindean** [comment]
eo :+1:

**Comment by lluchs** [comment]
`de` and `de-at` both look fine.

**Comment by mmozuras** [comment]
`lt` looks good :+1:

**Comment by kalehv** [comment]
`mr` :+1:

**Comment by kcthota** [comment]
Te - lgtm

**Comment by flakerimi** [comment]
sq (y)

**Comment by jfroffice** [comment]
```fr``` looks good :+1:

**Comment by miodragnikac** [comment]
Montenegrin `me` looks good. :+1:

**Comment by eillarra** [comment]
Basque (eu) looks good.

**Comment by sirn** [comment]
`th` looks good. :+1:

**Comment by gaspard** [comment]
``fr-ch`` looks good. :+1:

**Comment by mirontoli** [comment]
cv is good.

**Comment by gurdiga** [comment]
`ro` looks good.

**Comment by muminoff** [comment]
`uz` looks fine. :+1:

**Comment by lukemcgregor** [comment]
en-nz looks fine :+1:

**Comment by thanyawzinmin** [comment]
my looks good (y)

**Comment by harpreetkhalsagtbit** [comment]
`pa-in` looks fine

**Comment by bangnk** [comment]
vi is ok.

**Comment by forabi** [comment]
`ar` looks good.

**Comment by juanghurtado** [comment]
`ca` and `gl` seems to be ok.

**Comment by frontyard** [comment]
Bosnian (bs) looks good.

**Comment by suvash** [comment]
`ne` looks fine.

**Comment by aliem** [comment]
`it` lgtm

**Comment by jonashdown** [comment]
gd looks good

**Comment by jbleduigou** [comment]
br looks good

**Comment by ragnar123** [comment]
I think there is some minor errors in *fo* locale. (s/síðstu/síðsti). Other
than that it looks good; i will just try to ask in a local language forum /
open source forum to make sure. And submit a patch when I know for sure.

Also it is Faroese, not Farose (on the front page).

Regards,
Ragnar

On Mon, Mar 14, 2016 at 12:05 PM, Jean-Baptiste Le Duigou <
notifications@github.com> wrote:

> br looks good
>
> On Mon, 14 Mar 2016 at 11:19 Jon Ashdown <notifications@github.com> wrote:
>
> > gd looks good
> >
> > —
> > Reply to this email directly or view it on GitHub
> > <https://github.com/moment/moment/pull/3034#issuecomment-196241424>.
>
> >
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/moment/moment/pull/3034#issuecomment-196260863>.
>



-- 
Vinarliga
Ragnar Johannesen

**Comment by skakri** [comment]
`lv` :shipit:

**Comment by wernerm** [comment]
`af` seems fine :+1:

**Comment by julionc** [comment]
`es` looks fine. :+1:

**Comment by amaranthrose** [comment]
`tlh` looks ok. :+1:

**Comment by chriscartlidge** [comment]
ie is fine.

**Comment by tyok** [comment]
`id` looks good :+1:

**Comment by fadsel** [comment]
`sw` looks neat :+1:

**Comment by mj1856** [comment]
@ragnar123 - I fixed the typo on the website.  I'll wait for your PR on the other bits.  Thanks!

**Comment by ragulka** [comment]
`et` looks good :+1:

**Comment by kyungw00k** [comment]
`ko` looks good.

**Comment by adambrunner** [comment]
`hu` looks good 👍🏼
Sorry for late response, was on holiday.

File: src/test/locale/bn.js [hunk 1]
```diff
@@ -150,14 +150,14 @@ test('fromNow', function (assert) {
...
-    assert.equal(moment(a).add({d: 1}).calendar(),       'আগামীকাল রাত ২:০০ সময়',  'tomorrow at the same time');
-    assert.equal(moment(a).subtract({h: 1}).calendar(),  'আজ রাত ১:০০ সময়',     'Now minus 1 hour');
-    assert.equal(moment(a).subtract({d: 1}).calendar(),  'গতকাল রাত ২:০০ সময়', 'yesterday at the same time');
+    var a = moment().hours(12).minutes(0).seconds(0);
+
+    assert.equal(moment(a).calendar(),                   'আজ দুপুর ১২:০০ সময়',       'today at the same time');
+    assert.equal(moment(a).add({m: 25}).calendar(),      'আজ দুপুর ১২:২৫ সময়',       'Now plus 25 min');
+    assert.equal(moment(a).add({h: 3}).calendar(),       'আজ দুপুর ৩:০০ সময়',        'Now plus 3 hours');
+    assert.equal(moment(a).add({d: 1}).calendar(),       'আগামীকাল দুপুর ১২:০০ সময়', 'tomorrow at the same time');
+    assert.equal(moment(a).subtract({h: 1}).calendar(),  'আজ দুপুর ১১:০০ সময়',       'Now minus 1 hour');
```
Comment (kaushikgandhi) [inline src/test/locale/bn.js]: This should be 'আজ সকাল ১১:০০ সময়'

**Comment by kaushikgandhi** [comment]
bn had one issue commented on the line 

Thanks

**Comment by kalehv** [comment]
@kaushikgandhi In Indian languages locales, Morning is treated as any hour < 10, it should be < 12 instead.

`mr` has same issue but that is because the way original code is. I am waiting to fix it in `mr` as well but that will have to be a separate PR since original code will need to change

Ref - https://github.com/moment/moment/blob/develop/src/locale/bn.js#L96

Forgive me if I am incorrect, I do not understand Bengali but I believe it is the same issue.

**Comment by mj1856** [comment]
@kaushikgandhi please open an issue or a PR for that.  thanks.

**Comment by tk120404** [comment]
`ta` looks good :+1:

**Comment by mj1856** [comment]
Thanks everyone that's responded so far.  I'd still like to hear from the following locale authors:

- [ ] az @topchiyev
- [ ] bo @vajradog
- [ ] dv @jawish
- [ ] en-au @jarcoal
- [ ] fi @bleadof
- [ ] fy @robin0van0der0v
- [ ] he @tomer @DevelopmentIL @TalAter
- [ ] hi @mayanksinghal
- [ ] hr @bmarkovic
- [ ] hy-am @armendarabyan
- [ ] is @hinrik
- [ ] ka @irakli-janiashvili
- [ ] kk @nurlan
- [ ] km @kruyvanna
- [ ] lo @ryanhart2
- [ ] ms @weldan
- [ ] ms-my @weldan
- [ ] nn @mechuwind
- [ ] pt @jalex79
- [ ] si @sampathsris
- [ ] sr @milan-j
- [ ] sr-cyrl @milan-j
- [ ] tl-ph @hagmandan
- [ ] tzl @robin0van0der0v
- [ ] tzm @vimdude
- [ ] tzm-latn @vimdude
- [ ] zh-cn @suupic
- [ ] zh-tw @zenozeng

**Comment by bleadof** [comment]
I've been super busy, but I will get back to this or tomorrow evening. :-)

**Comment by vimdude** [comment]
tzm looks good

**Comment by ichernev** [comment]
Merged in 69c259eb4925949af76d17ef216e72057ecd80dc

**Comment by mattjohnsonpint** [comment]
I tried to correct the history here, but it didn't help.  Nonetheless, this was indeed merged.
